### PR TITLE
Add PotionDataHolder interface

### PR DESCRIPTION
**Pull request by Pr0methean**
This will help simplify both the copy constructor of GlowMetaPotion and the initialization of GlowAreaEffectCloud (see https://github.com/GlowstoneMC/Glowstone/pull/614).

### CODE REVIEW

**Comment by Pr0methean** [comment]
This needs testing to ensure it builds and doesn't break Glowstone, but when running `mvn compile` (in the `Glowkit-Patched` subfolder) to test it myself, I get:
```
Failed to read artifact descriptor for co.aikar:fastutil-lite:jar:1.0: Could not transfer artifact co.aikar:fastutil-lite:pom:1.0 from/to glowstone (https://repo.glowstone.net/repository/internal/): peer not authenticated -> [Help 1]
```
I get the same error with `mvn -Dmaven.wagon.http.ssl.allowall=true compile`, so it probably isn't an SSL error that's fixable at my end (such as needing to add a new trusted root).

**Comment by mastercoms** [comment]
What version of Java and Maven are you using?

**Comment by Pr0methean** [comment]
Was using Java 7; successfully tested after installing Java 8.

File: Paper-API-Patches/0011-Add-PotionDataHolder-interface.patch [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
++     */
++    default void copyFrom(PotionDataHolder source) {
++        setBasePotionData(source.getBasePotionData());
++        setColor(source.getColor());
++        for (PotionEffect effect : source.getCustomEffects()) {
++            addCustomEffect(effect, true);
++        }
++    }
++}
+\ No newline at end of file
```
Comment (mastercoms) [inline Paper-API-Patches/0011-Add-PotionDataHolder-interface.patch]: Line here please

Comment (Pr0methean) [reply Paper-API-Patches/0011-Add-PotionDataHolder-interface.patch]: Done in 29df138

**Pull request merged**